client/controller/localized-links.js

```diff
@@ -28,7 +27,8 @@ export const setLocalizedCanonicalUrl = ( context, next ) => {
 		return;
 	}
 
-	const href = getLocalizedCanonicalUrl( context.originalUrl, getLocaleSlug() );
+	const langParamSlug = context.params.lang;
+	const href = getLocalizedCanonicalUrl( context.originalUrl, langParamSlug );
```

Comment:
Should we use `context.i18n.getLocaleSlug()` here instead of the lang URL parameter?

In theory, the URL parameter and the locale slug from i18n-calypso should always be the same, but if, for some reason, the two values are different, I think it might be a little more accurate to use the locale slug from the translation data used to render the page.

Reply:
Thanks for the suggestions here @yuliyan. I've swapped out the `context.params.lang` in favor of  `context.i18n.getLocaleSlug()`. Looks like this tests ok on my end with the same test plan previously. Would you mind taking another look?